lib/file-source/src/file_server.rs

```diff
@@ -299,6 +299,7 @@ where
                                 Ok(()) => {
                                     self.emitter.emit_file_deleted(&watcher.path);
                                     watcher.set_dead();
+                                    checkpoints.set_dead(file_id);
```

Comment:
Wouldn't the `watcher.set_dead()` that's already here cause it to be picked up by line 324?

Reply:
Aha, yes it would. Not sure why this was a separate path in my mind. Will remove!